File: src/com/watnapp/etipitaka/model/ModelBase.java
package com.watnapp.etipitaka.model;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;

abstract public class ModelBase {

	public ModelBase() {
		super();
	}

	abstract public void fromCursor(Cursor cursor, Context context);

	abstract public int getId();

	abstract public void setId(int id);

	abstract public ContentValues toContentValues();

}

File: app/src/main/java/com/watnapp/etipitaka/plus/model/ModelBase.kt
package com.watnapp.etipitaka.plus.model

import android.content.ContentValues
import android.content.Context
import android.database.Cursor

abstract class ModelBase {
    abstract fun fromCursor(cursor: Cursor, context: Context)
    abstract fun getId(): Int
    abstract fun setId(id: Int)
    abstract fun toContentValues(): ContentValues
}
